### PR TITLE
Add isOauth boolean parameter to the PersonalAccessToken type

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add `isOauth` boolean parameter to the `PersonalAccessToken` type to be able to filter tokens that was generated from oAuth. The `tokenName` parameter does not work for this case, because it is set from the original kubernetes secret object name, but not from the `che.eclipse.org/scm-personal-access-token-name` label.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse-che/che/issues/23241

### Is it tested? How?

1. Deploy the dashboard pull request image: `quay.io/eclipse/che-dashboard:pr-1269`
2. Configure [GitHub oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/), but do not authenticate it.
3. Go to the dashboard `User Preferences` -> `Personal Access Tokens` tab and add a GitHub PAT.
4. Go to the `User Preferences` -> `Git Services` tab and see that the GitHub authorisation is present but the authorisation status is empty and the kebab icon is disabled.

#### Release Notes

With this fix all git services items will have the authorized status, only if a related oAuth token was generated, but not with a manually added Personal Access Token.

### CODE REVIEW

**Comment by artaleks9** [comment]
Verified on Eclipse Che with `quay.io/eclipse/che-dashboard:pr-1269` - **the functionality works properly**

**Review by artaleks9 — APPROVED** [review]

File: packages/dashboard-frontend/src/store/GitOauthConfig/__tests__/actions.spec.ts [hunk 1]
```diff
@@ -294,32 +294,6 @@ describe('GitOauthConfig', () => {
...
-    });
-
-    it('should return false for invalid bitbucket-server token format', () => {
-      const gitProvider = `che-token-<user-id>-<invalid-hostname>`;
-      const result = isTokenGitProvider(gitProvider);
-      expect(result).toBe(false);
-    });
-  });
-
   describe('findUserToken', () => {
```
Comment (akurinnoy) [inline packages/dashboard-frontend/src/store/GitOauthConfig/__tests__/actions.spec.ts:297]:
```suggestion
  describe('findOauthToken', () => {
```

**Review by akurinnoy — APPROVED** [review]

**Comment by devspacesbuild** [comment]
Build [3.19](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [dashboard_3.x/598](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/598/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/598/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/598/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/dashboard_3.x/598/git)

**Comment by devspacesbuild** [comment]
Build [3.19](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [sync-to-downstream_3.x/8209](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8209/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8209/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8209/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8209/git)

**Comment by devspacesbuild** [comment]
Build [3.19](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8287](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/git)

**Comment by devspacesbuild** [comment]
Build [3.19](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8287](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/): [FAILURE](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8287/)<br /><blockquote>dashboard : 3.x :: <a href='https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=66077195'> Failed in  66077195</a> : BREW:BUILD/STATUS:UNKNOWN <br /> FAILURE:; copied to quay</blockquote>

**Pull request merged**